tests/integration/kubernetes/k8s-guest-pull-image-encrypted.bats

```diff
@@ -0,0 +1,133 @@
+}
+
+function create_pod_yaml_with_encrypted_image() {
+    image=$1
+
+    local CC_KBS_ADDR
+    export CC_KBS_ADDR=$(kbs_k8s_svc_http_addr)
+    kernel_params_annotation="io.katacontainers.config.hypervisor.kernel_params"
+    kernel_params_value+=" agent.guest_components_procs=confidential-data-hub"
+    kernel_params_value+=" agent.aa_kbc_params=cc_kbc::${CC_KBS_ADDR}"
+
+    # Note: this is not local as we use it in the caller test
+    kata_pod_with_encrypted_image="$(new_pod_config "$image" "kata-${KATA_HYPERVISOR}")"
+    set_node "${kata_pod_with_encrypted_image}" "$node"
+    set_container_command "${kata_pod_with_encrypted_image}" "0" "sleep" "30"
+
+    # Set annotation to pull image in guest
+    set_metadata_annotation "${kata_pod_with_encrypted_image}" \
+        "io.containerd.cri.runtime-handler" \
+        "kata-${KATA_HYPERVISOR}"
+    set_metadata_annotation "${kata_pod_with_encrypted_image}" \
+        "${kernel_params_annotation}" \
+        "${kernel_params_value}"
+
+    add_allow_all_policy_to_yaml "${kata_pod_with_encrypted_image}"
+}
+
+@test "Test that creating a container from an encrypted image, with no decryption key fails" {
+
+    # TODO - there is now delete KBS resource to ensure there is no key, so we need to keep
+    # this test running first to ensure that the KBS doesn't have the resource. An alternative
+    # is to run kbs_set_deny_all_resources, but we don't have a way to reset to the default
+    # policy, so for TEE tests we'd stay remaining with reject all, which could cause other
+    # subsequent tests to fail
+
+    create_pod_yaml_with_encrypted_image "${ENCRYPTED_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_encrypted_image}: $(cat ${kata_pod_with_encrypted_image})"
+
+    assert_pod_fail "${kata_pod_with_encrypted_image}"
+    assert_logs_contain "${node}" kata "${node_start_time}" 'failed to get decrypt key missing private key needed for decryption'
+}
+
+
+@test "Test that creating a container from an encrypted image, with correct decryption key works" {
+
+    setup_kbs_decryption_key "${DECRYPTION_KEY}" "${DECRYPTION_KEY_ID}"
+
+    create_pod_yaml_with_encrypted_image "${ENCRYPTED_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_encrypted_image}: $(cat ${kata_pod_with_encrypted_image})"
+
+    k8s_create_pod "${kata_pod_with_encrypted_image}"
+    echo "Kata pod test-e2e from encrypted image is running"
+}
+
+@test "Test that creating a container from an encrypted image, with incorrect decryption key fails" {
+
+    setup_kbs_decryption_key "anVua19rZXk=" "${DECRYPTION_KEY_ID}"
+
+    create_pod_yaml_with_encrypted_image "${ENCRYPTED_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_encrypted_image}: $(cat ${kata_pod_with_encrypted_image})"
+
+    assert_pod_fail "${kata_pod_with_encrypted_image}"
+    assert_logs_contain "${node}" kata "${node_start_time}" 'failed to get decrypt key missing private key needed for decryption'
```

Comment:
This log message is weird, isn't it? Because I would expect it to actually get the decrypt key but failing to actually decrypt the image.

Reply:
That is true. @Xynnn007 - do you have any thoughts about this? Anything I might be doing wrong in the test, or could the log message be better in the guest-components?

Reply:
Yes. I think it should  be promoted by guest components. An ancient issue for this https://github.com/confidential-containers/guest-components/issues/92 ; (

Reply:
hey, I created the issue and didn't even remember! I should find a doctor...

ok, so it is indeed failing to decrypt but just the message a little bit misleading.